pymaker/dsrmanager.py

```diff
@@ -0,0 +1,105 @@
+
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2020 kentonprescott
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from web3 import Web3
+from pymaker import Address, Contract, Transact
+from pymaker.dss import Pot, DaiJoin
+from pymaker.numeric import Wad, Rad
+from pymaker.token import DSToken
+
+
+class DsrManager(Contract):
+    """
+    A client for the `DsrManger` contract, which reduces the need for proxies
+    when interacting with the Pot contract.
+
+    Ref. <https://github.com/makerdao/dsr-manager/blob/master/src/DsrManager.sol>
+    """
+
+    abi = Contract._load_abi(__name__, 'abi/DsrManager.abi')
+    bin = Contract._load_bin(__name__, 'abi/DsrManager.bin')
+
+    def __init__(self, web3: Web3, address: Address):
+        assert isinstance(web3, Web3)
+        assert isinstance(address, Address)
+
+        self.web3 = web3
+        self.address = address
+        self._contract = self._get_contract(web3, self.abi, address)
+
+    def pot(self) -> Pot:
+        address = Address(self._contract.functions.pot().call())
+        return Pot(self.web3, address)
+
+    def dai(self) -> DSToken:
+        address = Address(self._contract.functions.dai().call())
+        return DSToken(self.web3, address)
+
+    def dai_adapter(self) -> DaiJoin:
+        address = Address(self._contract.functions.daiJoin().call())
+        return DaiJoin(self.web3, address)
+
+    def supply(self) -> Wad:
+        """Total supply of pie locked in Pot through DsrManager"""
+        return Wad(self._contract.functions.supply().call())
+
+    def pieOf(self, usr: Address) -> Wad:
```

Comment:
For consistency with (most of) this API and PEP8, please use [lowercase with underscores](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables) for method names.